### PR TITLE
refactor: get app directory from CLI

### LINKED ISSUE #436: Get app directory location from CLI
As a follow up after https://github.com/NativeScript/nativescript-dev-webpack/pull/433 is merged `nativescript-dev-webpack` will need information on where the project's app directory is located.

After https://github.com/NativeScript/nativescript-cli/pull/3356 is merged, however, this can no longer be hardcoded into `"app"` and would need to change.

### PR DESCRIPTION
**Pull request by sis0k0**
- Pass `sourceDir` and `appResourcesDir` when starting webpack from the `projectData` service.
- Read `sourceDir` and `appResourcesDir` in webpack configs from env object.
- Bootstrap CLI on postinstall to get the `appPath` value from the project data. Defaults to `app`.
- Pass appPath from webpack config to `getEntryModule`.

resolves #436

### CODE REVIEW

**Comment by sis0k0** [comment]
test

**Comment by sis0k0** [comment]
test

**Comment by sis0k0** [comment]
test

**Comment by sis0k0** [comment]
test

**Comment by sis0k0** [comment]
test

**Comment by sis0k0** [comment]
test

**Comment by sis0k0** [comment]
test

**Comment by sis0k0** [comment]
test

**Comment by sis0k0** [comment]
test

**Comment by sis0k0** [comment]
test

**Comment by sis0k0** [comment]
test

**Comment by sis0k0** [comment]
test

**Comment by sis0k0** [comment]
test

**Comment by sis0k0** [comment]
test

**Comment by sis0k0** [comment]
test

**Comment by sis0k0** [comment]
test

**Comment by sis0k0** [comment]
test

**Comment by sis0k0** [comment]
test

**Comment by sis0k0** [comment]
test

**Comment by sis0k0** [comment]
test

File: lib/compiler.js [hunk 1]
```diff
@@ -128,3 +120,34 @@ exports.runWebpackCompiler = function runWebpackCompiler(config, $mobileHelper,
...
+        $logger.warn("Stripping the snapshot flag. " +
+            "Bear in mind that snapshot is only available in release builds and " +
+            "is NOT available on Windows systems.");
+
+        hasLoggedSnapshotWarningMessage = true;
+    }
+}
+
+function getProjectDir($projectData) {
+    const projectDir = safeGet($projectData, "projectDir");
```
Comment (Mitko-Kerezov) [inline lib/compiler.js]: I don't believe this is necessary - CLI's `$projectData` always has `projectDir` property - if it doesn't the initial creation of the `$projectData` class would have failed long before we get here

File: projectHelpers.js [hunk 2]
```diff
@@ -3,6 +3,10 @@ const fs = require("fs");
 const semver = require("semver");
 const { EOL } = require("os");
 
+const { getProjectData, PROJECT_DATA_GETTERS } = require("./nsCliHelpers");
+
+const APP_DIR = "app";
```
Comment (Mitko-Kerezov) [inline projectHelpers.js]: You can reuse this constant [here](https://github.com/NativeScript/nativescript-dev-webpack/blob/2e9405615f71bba45ffcb242805141dbc10e3ac5/index.js#L14)

File: index.js [hunk 3]
```diff
@@ -12,23 +19,39 @@ if (isAngular({ projectDir: PROJECT_DIR })) {
     Object.assign(exports, require('./plugins/angular'));
 }
 
-exports.getEntryModule = function () {
-    const maybePackageJsonEntry = getPackageJsonEntry();
-    if (!maybePackageJsonEntry) {
-        throw new Error("app/package.json must contain a `main` attribute.");
+exports.getAotEntryModule = function (appDirectory = APP_DIR) {
+    const entry = getPackageJsonEntry(appDirectory);
+    const aotEntry = `${entry}.aot.ts`;
```
[the remark this answers is not in the document]
Comment (sis0k0) [reply index.js]: Yep, or Angular won't find it ;(

[the remark this answers is not in the document]
Comment (sis0k0) [reply index.js]: Yep, the Angular compiler needs a `ts` file.

**Comment by sis0k0** [comment]
test

**Comment by SvetoslavTsenov** [comment]
👍

**Comment by SvetoslavTsenov** [comment]
test

**Comment by SvetoslavTsenov** [comment]
test

**Comment by SvetoslavTsenov** [comment]
test

**Pull request merged**